libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -990,6 +990,8 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 		}
 		case FunctionType::Kind::Error:
 		{
+			// This case is part of the ``revert Error()`` path of the codegen.
+			// For ``require(bool, Error()`` refer to the ``Kind::Require`` case.
```

Comment:
```suggestion
			// This case is part of the ``revert Error()`` path of the codegen.
			// For ``require(bool, Error())`` refer to the ``Kind::Require`` case.
```
(Note parentheses :-))

I was about to suggest that it may in fact be safer to assert false here and move this to the revert visit - unfortunately since the reverts are statements and not expressions they don't have a visit, though, so it'd be rather annoying to do, so we can just keep it as is... it's not like if in the future we may allow "error constructor calls" elsewhere, we won't notice that this will unconditionally revert for all of them so far...

Reply:
Yup, Kamil and I spent quite a long time yesterday trying to figure out whether this could be de-duplicated easily, and came to more or less the same conclusion, hence the comment.

Reply:
Figures :-). Stop trying to refactor everything all the time in legacy code ;-P! But yeah, makes sense and it *was* a justified consideration here, but yeah, let's keep it at this.

Reply:
So what about the missing parenthesis?

Also, I'd say `revert <error>`/`require(bool, <error>)`, since it actually handles anything but `Error` :)